### ISSUE TITLE
[Bug] SuiClient.getNormalizedMoveModulesByPackage returns incorrect address in ABI for package addresses with leading zeros

### ISSUE THREAD

**Issue by nithilan4** [MystenLabs/sui#16083]
## Steps to Reproduce Issue

```js
const abiAddress = `0x02dab35...` // can be any package starting with a leading zero
const abi = await suiClient.getNormalizedMoveModulesByPackage({ package: abiAddress })
```

## Expected Result

All addresses in ABI should be the same as the specified package address, `0x02dab35`

## Actual Result

All addresses in ABI will be incorrect as `0x2dab354...`, without the leading zero.

## System Information

* OS: Ubuntu 22.04.2 LTS

**Comment by hayes-mysten** [comment]
@stefan-mysten I think this is an RPC issue, the SDK isn't modifying the address is the response.   I think the JSON RPC API does this with a lot of the addresses.  The most obvious one you'll see all over the place is "0x2", and I suspect that fixing this may also cause a lot of other things to break, because people likely are hard-coding checks against "0x2" in various places.

In the GraphQL API, we don't shorten addresses like that anymore, and you should always get the full address

**Comment by nithilan4** [comment]
I'm not sure about how it works internally, but I've heard that there are a lot of inconsistencies in this but the "rule" is that in the context of types the address generally don't have leading zeros (0x000....01::xx::yy becomes 0x1::xx::yy), but when in the context of addresses it is usually the full address.

**Comment by amnn** [comment]
Indeed, JSON-RPC introduces a number of special cases around package IDs (the one I remember is that well-known packages with short IDs, like the system packages, were displayed with trailing zeroes stripped), and some endpoints were returning all IDs with leading zeroes stripped.

Note that stripping leading zeroes does not change the addresses identity, `0x0000002` and `0x2` are the same address from the perspective of the chain (this is true for packages and for addresses), but we did recognise that when handling addresses programmatically, it is useful to be able to rely on the output format, which is why as @hayes-mysten mentioned, GraphQL always returns the canonical address with a leading `0x` and all its leading zeroes. It also accepts addresses without their leading zeroes for convenience.

The issue now with making a change to JSON-RPC to canonicalise addresses is that if we accept that people are treating the literal string representing the address as its identity (which means that presence of lack of leading zeroes has an effect) then we also need to acknowledge that people may equally be depending on the current behaviour that leading zeroes are stripped, and changing this will break working code.
